model_data/model_data_extract.py

```diff
@@ -124,6 +123,32 @@ def extract_opa_data(spark: SparkContext, save_path: str, fyear: int) -> None:
         .withColumn("is_wla", F.lit(True))
     )
 
+    inequalities = (
+        spark.read.parquet("/Volumes/nhp/inputs_data/files/dev/inequalities.parquet")
+        .filter(F.col("fyear") == fyear)
+        .select("provider", "sushrg_trimmed")
+        .withColumnRenamed("provider", "dataset")
+        .distinct()
+    )
+
+    # We don't want to keep sushrg_trimmed and imd_quintile if not in inequalities parquet file
+    opa_collapse = (
+        opa.join(inequalities, how="anti", on=["dataset", "sushrg_trimmed"])
+        .withColumn("sushrg_trimmed", F.lit(None))
+        .withColumn("imd_quintile", F.lit(None))
+    )
+    opa = (
+        opa_collapse.unionByName(
+            opa.join(inequalities, how="inner", on=["dataset", "sushrg_trimmed"])
+        )
+        .groupBy(opa.drop("index", "attendances", "tele_attendances").columns)
+        .agg(
+            F.sum("attendances").alias("attendances"),
+            F.sum("tele_attendances").alias("tele_attendances"),
+        )
+        .withColumn("index", F.expr("uuid()"))
+    )
```

Comment:
I think that it would be a bit clearer to read and understand at a later date if we split the collapse/don't collapse up like below. We only need to aggregate on the "collapse" section, so handle there. We can also utilise the existing index values by selecting the minimum in the group rather than re-assigning

```suggestion
    opa_collapse = (
        opa.join(inequalities, how="anti", on=["dataset", "sushrg_trimmed"])
        .withColumn("sushrg_trimmed", F.lit(None))
        .withColumn("imd_quintile", F.lit(None))
        .groupBy(opa.drop("index", "attendances", "tele_attendances").columns)
        .agg(
            F.sum("attendances").alias("attendances"),
            F.sum("tele_attendances").alias("tele_attendances"),
            F.min("index").alias("index")
        )
    )

    opa_dont_collapse = (
        opa.join(inequalities, how="semi", on=["dataset", "sushrg_trimmed"])
    )

    opa = DataFrame.unionByName(opa_collapse, opa_dont_collapse)
```